data/items.ts

```diff
@@ -2587,7 +2587,7 @@ export const Items: {[itemid: string]: ItemData} = {
 		onDamagingHit(damage, target, source, move) {
 			if (move.category === 'Physical') {
 				if (target.eatItem()) {
-					this.damage(source.baseMaxhp / 8, source, target);
+					this.damage(source.baseMaxhp / (this.effectData.ripe ? 4 : 8), source, target);
```

Comment:
```suggestion
					this.damage(source.baseMaxhp / (target.hasAbility('ripen') ? 4 : 8), source, target);
```
Doesn't this accomplish the same thing? I know you mention flexibility but I'm not sure theoretical future uses are worth adding EffectState all over the place in this case.

Reply:
Bump @urkerab, I am also generally a fan of just hardcoding the Ripen Ability for Jaboca and Rowap.